### PR TITLE
Replacing custom interface with stock openzeppelin IERC721.sol

### PR DESCRIPTION
**Pull request by nfurfaro**
# Description
As per the title.
- changed `IPublicLock.sol` to abstract prevent compilation errors
- modified a few functions' visibility from `external` to `public` to prevent compilation errors
- added 2 getter functions (`ownerOf` and `totalSupply`) and changed existing to `ownerOf` and `_totalSupply` tp prevent duplicate declaration error. 
- added one extra check in `function ownerOf` based on the openzeppelin implementation of the same function.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by NickCuso** [comment]
Tests are really not happy right now ;)

**Comment by nfurfaro** [comment]
Huh. maybe something going on with solium... indentation looks fine to me. Will see if I can make it happy again.

**Comment by NickCuso** [comment]
> Huh. maybe something going on with solium... indentation looks fine to me. Will see if I can make it happy again.

Note that both the tests AND lint are failing.

**Comment by NickCuso** [comment]
This can be closed, right?  Was replaced by another PR - re-open if I missed something.

**Comment by nfurfaro** [comment]
Reopening as I have this working locally now.

**Comment by NickCuso** [comment]
Sorry about that, at a glance I thought https://github.com/unlock-protocol/unlock/pull/5383 replaced this.

Let me know once it's ready for review

**Comment by nfurfaro** [comment]
@NickCuso When you have time could you give this a look over? thx

File: smart-contracts/contracts/interfaces/IPublicLock.sol [hunk 1]
```diff
@@ -1,21 +1,20 @@
 pragma solidity ^0.5.0;
 
-import './IERC721.sol';
-import './IERC721Enumerable.sol';
+import '@openzeppelin/contracts-ethereum-package/contracts/token/ERC721/IERC721Enumerable.sol';
 
+import '@openzeppelin/contracts-ethereum-package/contracts/ownership/Ownable.sol';
```
Comment (NickCuso) [inline smart-contracts/contracts/interfaces/IPublicLock.sol]: This import can be removed it seems

File: smart-contracts/contracts/mixins/MixinApproval.sol [hunk 2]
```diff
@@ -40,7 +38,7 @@ contract MixinApproval is
     require(
       isKeyOwner(_tokenId, msg.sender) ||
         _isApproved(_tokenId, msg.sender) ||
-        isApprovedForAll(ownerOf[_tokenId], msg.sender),
+        isApprovedForAll(_ownerOf[_tokenId], msg.sender),
```
Comment (NickCuso) [inline smart-contracts/contracts/mixins/MixinApproval.sol]: Hmm if we are switching to the ZOS pattern than the data would be private and we use the public function instead.

Comment (NickCuso) [reply smart-contracts/contracts/mixins/MixinApproval.sol]: They also have a require statement in the ownerOf implementation... not sure if that we should too though

Comment (nfurfaro) [reply smart-contracts/contracts/mixins/MixinApproval.sol]: Yeah, I can review through and try to use the public function more consistently. As for the require, we do have it, but it's in the `isKey` modifier which is used in our `ownerOf` implementation.

File: smart-contracts/contracts/interfaces/IPublicLock.sol (hunk 1 above)
Comment (nfurfaro) [reply smart-contracts/contracts/interfaces/IPublicLock.sol]: yup. done.

File: smart-contracts/contracts/mixins/MixinApproval.sol (hunk 2 above)
Comment (nfurfaro) [reply smart-contracts/contracts/mixins/MixinApproval.sol]: k, I've now used the public function everywhere I can, and made the data private. The only exceptions are for write operations.

Comment (NickCuso) [reply smart-contracts/contracts/mixins/MixinApproval.sol]: didn't mean to complicate things :)  i don't feel strongly either way if you want to revert to what you had

**Comment by nfurfaro** [comment]
Ha! well, that didn't work.

**Comment by NickCuso** [comment]
heh yea didn't mean to complicate things :) i don't feel strongly either way if you want to revert to what you had

**Comment by nfurfaro** [comment]
I will revert the last commit which switched from internal to private. I think the rest should be good.

**Comment by nfurfaro** [comment]
ok, reverted to my earlier PR.

**Comment by nfurfaro** [comment]
@NickCuso If you're ok with the approach, I'd like to merge this. We can certainly look at further tweaks later, but right now the small changes discussed above break so many tests, I'm not sure the value is there in pursuing them atm.

File: smart-contracts/contracts/PublicLock.sol [hunk 3]
```diff
@@ -1,11 +1,9 @@
 pragma solidity 0.5.13;
 
-import './interfaces/IERC721.sol';
-import './interfaces/IERC721Enumerable.sol';
+
 import './interfaces/IPublicLock.sol';
 import '@openzeppelin/contracts-ethereum-package/contracts/ownership/Ownable.sol';
 import '@openzeppelin/contracts-ethereum-package/contracts/introspection/ERC165.sol';
-import '@openzeppelin/upgrades/contracts/Initializable.sol';
```
Comment (NickCuso) [inline smart-contracts/contracts/PublicLock.sol]: Why remove initializable?

Comment (nfurfaro) [reply smart-contracts/contracts/PublicLock.sol]: just some cleanup... `PublicLock.sol` already inherits `Initializable.sol` via both `IPublicLock.sol` and `MixinERC721Enumerable.sol`

**Pull request merged**